Mathlib/Topology/LocalAtTarget.lean

```diff
@@ -64,16 +64,13 @@ alias ClosedEmbedding.restrictPreimage := Set.restrictPreimage_closedEmbedding
 #align closed_embedding.restrict_preimage ClosedEmbedding.restrictPreimage
 
 theorem Set.restrictPreimage_isClosedMap (s : Set β) (H : IsClosedMap f) :
-    IsClosedMap (s.restrictPreimage f) := by
-  rintro t ⟨u, hu, e⟩
-  refine' ⟨⟨_, (H _ (IsOpen.isClosed_compl hu)).1, _⟩⟩
-  rw [← (congr_arg HasCompl.compl e).trans (compl_compl t)]
-  simp only [Set.preimage_compl, compl_inj_iff]
-  ext ⟨x, hx⟩
-  suffices (∃ y, y ∉ u ∧ f y = x) ↔ ∃ y, y ∉ u ∧ f y ∈ s ∧ f y = x by
-    simpa [Set.restrictPreimage, ← Subtype.coe_inj]
-  exact ⟨fun ⟨a, b, c⟩ => ⟨a, b, c.symm ▸ hx, c⟩, fun ⟨a, b, _, c⟩ => ⟨a, b, c⟩⟩
-#align set.restrict_preimage_is_closed_map Set.restrictPreimage_isClosedMap
+    IsClosedMap (s.restrictPreimage f) :=
+  fun t ⟨u, hu, e⟩ =>  ⟨(f '' uᶜ)ᶜ, isOpen_compl_iff.2 (H uᶜ (IsOpen.isClosed_compl hu)),
+    by simp [e, ← image_restrictPreimage]⟩
```

Comment:
This works, but is in some sense not the "morally correct" way to do this, since you're abusing the actual definition of `IsClosed` in terms of `IsOpen`. Could you use [isClosed_induced_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Order.html#isClosed_induced_iff) instead? The proof may get a bit longer, but I'd argue that it's a bit cleaner.

Reply:
I corrected this proof and the next one as you said. It is indeed slightly longer, but I am not sure that I wrote the most efficient proofs.